### PR TITLE
Adding Code coverage to ublas

### PR DESCRIPTION
**Pull request by coder3101**
This PR brings the code-coverage support on codecov for ublas.
- Currently, all coverage reports are gathered for every unit test executed. If we want to only limit the coverage reporting to some modules (say tensor). Please remove the file`--coverage` linker and compiler flag of gcc from Jamfile of respective tests. 
- PR also has some minor formatting here in tensor.hpp file. (No big change).
- It also adds .gitignore file for excluding some IDE related folders (vscode in this case)

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ branches:
     - master
     - develop
     - /feature\/.*/
+    - expr_template
```
Comment (bassoy) [inline .appveyor.yml]: This should not be merged into the develop branch

File: .travis.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ branches:
     - develop
     - doc
     - ci
+    - expr_template
```
Comment (bassoy) [inline .travis.yml]: This should not be merged into the develop branch

Comment (coder3101) [reply .travis.yml]: This is outdated. In the last commit named I have removed this. Please check the final changes

Comment (bassoy) [reply .travis.yml]: yes. I now have seen it.

**Comment by coder3101** [comment]
Those are outdated commits that were previously there. Last commit b27cad3 fixes this already.

**Comment by coder3101** [comment]
I had to to a force-push because of squashing the commits into one commit.

**Pull request merged**